packages/toolpad-app/src/toolpad/AppEditor/PageEditor/ComponentCatalog/ComponentCatalog.tsx

```diff
@@ -0,0 +1,270 @@
+          <Box sx={{ width: 270, height: '100%', overflow: 'auto' }}>
+            <Box display="grid" gridTemplateColumns="1fr 1fr 1fr" gap={1} padding={1}>
+              {Object.entries(toolpadComponents).map(([componentId, componentType]) => {
+                invariant(componentType, `No component definition found for "${componentId}"`);
+                return componentType.builtIn && !componentType.system ? (
+                  <ComponentCatalogItem
+                    key={componentId}
+                    id={componentId}
+                    draggable
+                    onDragStart={handleDragStart(componentId)}
+                    displayName={componentType.displayName}
+                    builtIn={componentType.builtIn}
+                    kind={'builtIn'}
+                  />
+                ) : null;
+              })}
+            </Box>
+
+            <Box px={2} pb={0} display="flex" flexDirection={'row'} justifyContent="space-between">
+              <Typography variant="overline">Custom Components</Typography>
+              <IconButton
+                aria-label="Expand custom components"
+                sx={{
+                  p: 0,
+                  cursor: 'pointer',
+                  transform: `rotate(${openCustomComponents ? 180 : 0}deg)`,
+                  transition: 'all 0.2s ease-in',
+                }}
+                onClick={() => setOpenCustomComponents((prev) => !prev)}
+              >
+                <ArrowDropDownSharpIcon />
+              </IconButton>
+            </Box>
+            <Collapse in={openCustomComponents} orientation={'vertical'}>
+              <Box display="grid" gridTemplateColumns="1fr 1fr 1fr" gap={1} padding={1} pt={0}>
+                {Object.entries(toolpadComponents).map(([componentId, componentType]) => {
+                  invariant(componentType, `No component definition found for "${componentId}"`);
+                  return !componentType.builtIn ? (
+                    <ComponentCatalogItem
+                      key={componentId}
+                      id={componentId}
+                      draggable
+                      onDragStart={handleDragStart(componentId)}
+                      displayName={componentType.displayName}
+                      kind={'custom'}
+                    />
+                  ) : null;
+                })}
+                <ComponentCatalogItem
+                  id={'CreateNew'}
+                  displayName={'Create'}
+                  kind={'create'}
+                  onClick={handleCreateCodeComponentDialogOpen}
+                />
+              </Box>
+            </Collapse>
+
+            <Box padding={1}>
+              <Box
+                sx={{
+                  py: 2,
+                  px: 1,
+                  borderWidth: 1,
+                  borderStyle: 'inset',
+                  borderColor: 'divider',
+                  backgroundColor: 'grey.200',
```

Comment:
Will this work with a dark theme?

Reply:
I've made some updates to this PR to handle the new theme on the catalog; I'll add some changes to the catalog in https://github.com/mui/mui-toolpad/pull/988 as well so they get applied when it gets merged

Reply:
I'm mainly wondering whether it's possible to make this more theme-agnostic? Instead of specifying a specific color, rather darken/lighten the background. Or darken/lighten a theme color.

Testing whether a generic dark theme produces acceptable results is just a matter of switching the palette mode.

Reply:
Agreed, I've made a change to use `darken` which is theme agnostic and does the job:


https://user-images.githubusercontent.com/19550456/196383262-fdd83a15-7b2a-4a83-8db6-aaa63a4af3a5.mov